llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2755,17 +2755,22 @@ void InnerLoopVectorizer::fixupIVUsers(PHINode *OrigPhi,
       if (isa_and_nonnull<FPMathOperator>(II.getInductionBinOp()))
         B.setFastMathFlags(II.getInductionBinOp()->getFastMathFlags());
 
-      Value *CountMinusOne = B.CreateSub(
-          VectorTripCount, ConstantInt::get(VectorTripCount->getType(), 1));
-      CountMinusOne->setName("cmo");
-
       VPValue *StepVPV = Plan.getSCEVExpansion(II.getStep());
       assert(StepVPV && "step must have been expanded during VPlan execution");
       Value *Step = StepVPV->isLiveIn() ? StepVPV->getLiveInIRValue()
                                         : State.get(StepVPV, VPLane(0));
-      Value *Escape =
-          emitTransformedIndex(B, CountMinusOne, II.getStartValue(), Step,
-                               II.getKind(), II.getInductionBinOp());
+      Value *Escape = nullptr;
```

Comment:
One of the things that `emitTransformedIndex` does is add a cast between `CountMinusOne` and the type of `Step`. Is it worth adding an assert that the types match?

Reply:
`CountMinusOne` has the same type as `VectorTripCount` which may be different to the type of `Step`. `EndValue` should always have the same type as the `Step`, as it is computed using it. There should already be asserts when creating the expression that the types match which should catch any differences here.